### DISCUSSION TITLE
Using Renovate with Next.js

### DISCUSSION THREAD

**Discussion by milovangudelj** [vercel/next.js#46858] [Help] [answered]
### Summary

I'm using the Renovate bot to keep up with dependency updates, but this is leading to a hell of rebuilds whenever more than one dependency needs a bump at once.

Whenever Renovate detects a newer version of one of your application's dependecies it creates a new pull request with the update. 
This causes an issue though since if there is more than one open pr each one gets built on Vercel, and then as soon as one of them is merged, a new build is triggered for each remaining open pr plus a new build on the `master` branch with the merged changes.

This way you end up with $\frac{n^2 + 3n}{2}$ builds, wich is a pretty significant amount of build time.
```
For example:

  2 prs -> 5  builds
  3 prs -> 9  builds
  4 prs -> 14 builds
  5 prs -> 20 builds
```

Is there a better way of using Renovate with Vercel and Next.js? Am I doing something wrong or is this just the way it works and there's nothing I can do about it?

Let me know if you need any more information on this or on my configuration.

**Comment by joulev** [comment, marked as answer]
You can use [`commitBody`](https://docs.renovatebot.com/configuration-templates/#commit-message) for example to add `skip ci` to, well, skip CI/CD.

But I'd recommend grouping non-major version upgrades together to one PR only, and let it build like the default way. I would never be confident to merge any PR unless I have *at least* received confirmation that it can build well when merged.

If that is still too many builds for you, you can consider decreasing the frequency of PRs as well (to weekly/monthly for example).

**Reply by milovangudelj** [reply]
Thanks for the suggestions! I will try them out and see what's best for me :+1:
